src/building/count.c

```diff
@@ -374,3 +377,129 @@ int building_count_fort_type_total(figure_type type)
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_plaza(void)
+{
+    int total = 0;
+    int grid_offset = map_data.start_offset;
+    for (int y = 0; y < map_data.height; y++, grid_offset += map_data.border_size) {
+        for (int x = 0; x < map_data.width; x++, grid_offset++) {
+            if (map_terrain_is(grid_offset, TERRAIN_ROAD) && map_property_is_plaza_earthquake_or_overgrown_garden(grid_offset)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_gardens(int overgrown)
+{
+    int total = 0;
+    int grid_offset = map_data.start_offset;
+    for (int y = 0; y < map_data.height; y++, grid_offset += map_data.border_size) {
+        for (int x = 0; x < map_data.width; x++, grid_offset++) {
+            if (map_terrain_is(grid_offset, TERRAIN_GARDEN)) {
+                int is_overgrown = map_property_is_plaza_earthquake_or_overgrown_garden(grid_offset) != 0;
+                if (is_overgrown == overgrown) {
+                    total++;
+                }
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_roads_area(int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+
+            if (map_terrain_is(grid_offset, TERRAIN_ROAD)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_highway_area(int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+
+            if (map_terrain_is(grid_offset, TERRAIN_HIGHWAY)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_plaza_area(int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+
+            if (map_terrain_is(grid_offset, TERRAIN_ROAD) && map_property_is_plaza_earthquake_or_overgrown_garden(grid_offset)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_gardens_area(int minx, int miny, int maxx, int maxy, int overgrown)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+            
+            if (map_terrain_is(grid_offset, TERRAIN_GARDEN)) {
+                int is_overgrown = map_property_is_plaza_earthquake_or_overgrown_garden(grid_offset) != 0;
+                if (is_overgrown == overgrown) {
+                    total++;
+                }
+            }
+        }
+    }
+    return total;
+}
```

Comment:
```suggestion
}

```

(Newline at end of file)